core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -885,31 +884,39 @@ impl<R: WriteableRocks> StateManagerDatabase<R> {
     fn ensure_historical_substate_values(&self) {
         let db_context = self.open_rw_context();
         let extension_data_cf = db_context.cf(ExtensionsDataCf);
-        let values_associated_since = extension_data_cf
-            .get(&ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince)
-            .map(|bytes| scrypto_decode::<StateVersion>(&bytes).unwrap());
+        let status = extension_data_cf
+            .get(&ExtensionsDataKey::StateHashTreeAssociatedValuesStatus)
+            .map(|bytes| {
+                scrypto_decode::<VersionedStateHashTreeAssociatedValuesStatus>(&bytes)
+                    .unwrap()
+                    .into_latest()
+            });
 
         if self.config.enable_historical_substate_values {
-            if let Some(version) = values_associated_since {
-                info!("Historical Substate values enabled since {:?}", version);
+            if let Some(status) = status {
+                info!("Historical Substate values enabled since {:?}", status);
             } else {
                 let current_version = self.max_state_version();
                 info!(
                     "Enabling historical Substate values at {:?}",
                     current_version
                 );
                 self.populate_hash_tree_associated_substate_values(current_version);
+                let status = StateHashTreeAssociatedValuesStatusV1 {
+                    values_associated_from: current_version,
+                };
                 extension_data_cf.put(
-                    &ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince,
-                    &scrypto_encode(&current_version).unwrap(),
+                    &ExtensionsDataKey::StateHashTreeAssociatedValuesStatus,
+                    &scrypto_encode(&VersionedStateHashTreeAssociatedValuesStatus::from(status))
```

Comment:
Minor - but I'm surprised we have to Scrypto Encode here - can't we use a better codec? :D

Reply:
what do you mean by "better codec"?

if you mean "use the `TypedCf` infra", then it's not that easily available for the "extensions data" CF, due to the inherently "polymorphic" nature of this "table" :(

Reply:
Ahh yes, quite right!